service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/IdMapper.java

```diff
@@ -39,23 +38,43 @@ public class IdMapper {
         ResourceType.PractitionerRole.name());
 
     public String getOrNew(ResourceType resourceType, IdType id) {
-        return getOrNew(buildReference(resourceType, id), true);
+        return getOrNew(resourceType, id, true);
     }
 
     public String newId(ResourceType unmappedResource, IdType id) {
-        return getOrNew(buildReference(unmappedResource, id), false);
+        return getOrNew(unmappedResource, id, false);
     }
 
     public String getOrNew(Reference reference) {
-        return getOrNew(reference, false);
+        return getOrNewFromReference(reference);
     }
 
-    public String getOrNew(Reference reference, Boolean isResourceMapped) {
+    public String getOrNewFromReference(Reference reference) {
         if (NOT_ALLOWED.contains(reference.getReferenceElement().getResourceType())) {
             throw new EhrMapperException("Not allowed to use agent-related resource with IdMapper");
         }
 
-        MappedId defaultResourceId = new MappedId(randomIdGeneratorService.createNewId(), isResourceMapped);
+        var referenceId = reference.getReferenceElement().getIdPart();
```

Comment:
I think introducing a variable for reference.getReferenceElement() would give more readability.

Reply:
Fair point, will update that now